File: NasleGhalam.ViewModels/Topic/TopicViewModel.cs
﻿using System.ComponentModel.DataAnnotations;
using NasleGhalam.ViewModels.Lesson;

namespace NasleGhalam.ViewModels.Topic
{
    public class TopicViewModel
    {
        [Display(Name = "")]
        public int Id { get; set; }


        [Display(Name = "")]
        public string Title { get; set; }


        [Display(Name = "")]
        public int ExamStock { get; set; }


        [Display(Name = "")]
        public int ExamStockSystem { get; set; }


        [Display(Name = "")]
        public short Importance { get; set; }


        [Display(Name = "")]
        public bool IsExamSource { get; set; }


        [Display(Name = "")]
        public int LookupId_HardnessType { get; set; }


        [Display(Name = "")]
        public int LookupId_AreaType { get; set; }


        [Display(Name = "")]
        public bool IsActive { get; set; }


        [Display(Name = "")]
        public int LessonId { get; set; }

        //public LessonViewModel Lesson { get; set; }

        public int? ParentTopicId { get; set; }

        //public TopicViewModel TopicViewModel { get; set; }



    }
}

File: NasleGhalam.ClientApp/src/viewModels/topicViewModel.js
import {
  required,
  displayName,
  maxLength,
  numeric,
  requiredDdl
} from 'plugins/vuelidate';

export default {
  topicObj: {
    Title: {
      displayName: displayName('عنوان'),
      maxLength: maxLength(50),
      required
    },
    ExamStock: {
      displayName: displayName('سهمیه در کنکور'),
      numeric,
      required
    },
    ExamStockSystem: {
      displayName: displayName('سهمیه در کنکور سیستمی')
      // numeric,
      // required
    },
    Importance: {
      displayName: displayName('ضریب اهمیت'),
      numeric,
      required
    },
    IsExamSource: {
      displayName: displayName('مبحث کنکوری'),

      required
    },
    LookupId_HardnessType: {
      displayName: displayName('درجه سختی'),
      numeric,
      required
    },
    LookupId_AreaType: {
      displayName: displayName('حیطه مبحث'),
      numeric,
      required
    },
    IsActive: {
      displayName: displayName('فعال'),

      required
    },
    ParentTopicId: {
      displayName: displayName('مبحث پدر')
      // numeric,
      // required
    },
    EducationGroup_LessonId: {
      displayName: displayName('درس')
      // numeric,
      // required
    },
    EducationGroupId: {
      displayName: displayName('گروه آموزشی')
      // numeric,
      // required
    },
    LookupId_Nezam: {
      displayName: displayName('نظانم (اختیاری)')
    },
    GradeId: {
      displayName: displayName('مقطع (اختیاری)')
    },
    GradeLevelId: {
      displayName: displayName('پایه (اختیاری)')
    }
  }
};
